bin/brew

```diff
@@ -45,5 +85,77 @@ fi
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_RUBY_PATH
+export HOMEBREW_PREFIX
+export HOMEBREW_REPOSITORY
+export HOMEBREW_LIBRARY
+export HOMEBREW_CELLAR
+
+if [ -n "$HOMEBREW_OSX" ]
+then
+  if [ "$(xcode-\select --print-path)" = "/" ]
+  then
+    cat >&2 <<EOS
+Your xcode-select path is currently set to '/'.
+This causes the 'xcrun' tool to hang, and can render Homebrew unusable.
+If you are using Xcode, you should:
+  sudo xcode-select -switch /Applications/Xcode.app
+Otherwise, you should:
+  sudo rm -rf /usr/share/xcode-select
+EOS
+    exit 1
+  fi
+
+  XCRUN_OUTPUT="$(/usr/bin/xcrun clang 2>&1)"
+  XCRUN_STATUS="$?"
+
+  if [ "$XCRUN_STATUS" != "0" ] && [[ "$XCRUN_OUTPUT" =~ license ]]
+  then
+    cat >&2 <<EOS
+You have not agreed to the Xcode license. Please resolve this by running:
+  sudo xcodebuild -license
+EOS
+    exit 1
+  fi
+fi
+
+# Many Pathname operations use getwd when they shouldn't, and then throw
+# odd exceptions. Reduce our support burden by showing a user-friendly error.
+if ! [ -d "$(pwd)" ]
+then
+  echo "The current working directory doesn't exist, cannot proceed."
+  exit 1
+fi
+
+for i in "$@"
+do
+  [[ "$i" =~ ^- ]] && continue
```

Comment:
> This only applies to Bash commands for now, […]

Sorry to disagree again, but no. When this code is running, it's not clear yet what the command is and thus also equally unclear how it is implemented. It changes the interpretation of arguments for *everything* that is invoked via `brew`. (But I agree that most examples that show this have to be constructed and are unlikely to affect normal usage.)